library/psa_crypto.c

```diff
@@ -2412,10 +2454,16 @@ psa_status_t psa_mac_verify_finish( psa_mac_operation_t *operation,
     psa_status_t abort_status = PSA_ERROR_CORRUPTION_DETECTED;
 
     if( operation->id == 0 )
-        return( PSA_ERROR_BAD_STATE );
+    {
+        status = PSA_ERROR_BAD_STATE;
+        goto cleanup;
```

Comment:
I'd move to `exit` here as well just to align all functions we are working on in this PR.

Reply:
Done